### DISCUSSION TITLE
How to implement this feature ?

### DISCUSSION THREAD

**Discussion by FgrReloaded** [microsoft/vscode-discussions#1280] [Extension Development QnA] [answered]
_Hey, I am currently new to vs code extension development.
I am working on a extension and I want to implement a inline input box with few buttons._

**As shown in image below**:
![Screenshot 2024-07-20 093056](https://github.com/user-attachments/assets/2166408d-6328-43c0-9c7c-c36abb53262c)

How can I add such input box after executing any specific commands ?

**Comment by alefragnani** [comment, marked as answer]
Hi @FgrReloaded ,

There is no API available for this feature yet, at least not  _finalized_. Which means you can't _publish_ an extension that uses it.

You may take a look at the `editorInsets` Proposed API (https://github.com/microsoft/vscode/blob/main/src/vscode-dts/vscode.proposed.editorInsets.d.ts), with a few details on this (https://github.com/microsoft/vscode/issues/85682). Personally, I didn't play with this API yet, so I'm not entirely sure how it works, and how far you can go with it.

Hope this helps
